### PR TITLE
Fixes the rpmdb bootstrap management for DNF on SUSE

### LINKED ISSUE #1669: rpmdb is broken in opensuse/tumbleweed-dnf
## Problem description
openSUSE images produced by KIWI using DNF lack a correctly populated rpmdb for some reason. This leads to package management being completely broken in these images.

## Expected behavior

1. `rpm -qa` returns a full list of packages
2. `dnf --version` returns both the version and info from rpmdb, like it does on the Fedora Rawhide container.

```
ngompa@localhost ~> podman run --pull=always --rm -it registry.fedoraproject.org/fedora:rawhide
Trying to pull registry.fedoraproject.org/fedora:rawhide...
Getting image source signatures
Copying blob ad16fb0e8007 done
Copying config 739bc13b58 done
Writing manifest to image destination
Storing signatures
[root@b445671f5a5c /]# dnf --version
4.5.2
  Installed: dnf-0:4.5.2-1.fc34.noarch at Mon Dec 21 09:56:38 2020
  Built    : Fedora Project at Thu Dec  3 16:19:10 2020

  Installed: rpm-0:4.16.1.2-1.fc34.x86_64 at Mon Dec 21 09:56:37 2020
  Built    : Fedora Project at Wed Dec 16 12:57:28 2020
```

## Steps to reproduce the behavior
```
ngompa@localhost ~> podman run --pull=always --rm -it registry.opensuse.org/opensuse/tumbleweed-dnf
Trying to pull registry.opensuse.org/opensuse/tumbleweed-dnf:latest...
Getting image source signatures
Copying blob 3acf88ce06c4 [--------------------------------------] 0.0b / 0.0b
Copying config 60322246ca done
Writing manifest to image destination
Storing signatures
944335ab1164:/ # rpm -qa
gpg-pubkey-307e3d54-5aaa90a5
gpg-pubkey-39db7c82-5f68629b
gpg-pubkey-3dbdc284-53674dd4
944335ab1164:/ # dnf --version
4.5.2
```

## OS and Software information
+ KIWI version: 9.22.1 (version in openSUSE Tumbleweed at this writing)
+ Operating system: openSUSE Tumbleweed (20201218)
+ OBS version: 2.11~alpha.20201218T151113.0e3def877c (openSUSE Build Service on build.opensuse.org)

### PR DESCRIPTION
**Pull request by davidcassany**
This commit makes sure that the compatibility symlink for
/var/lib/rpm is created when the host rpmdb path is set to something
different. This fixes a mismatch on bootstrapping SUSE using DNF.

Fixes #1669

### CODE REVIEW

**Comment by davidcassany** [comment]
@Conan-Kudo this looks safe to me, however I'd appreciate if you could run some Fedora test from a Fedora host. So far I successfully tested from my Leap host a Tumbleweed build using DNF as package manager and a Fedora 32 build also from my Leap host.

File: test/unit/repository/dnf_test.py [hunk 1]
```diff
@@ -163,9 +163,14 @@ def test_add_repo_inside_buildservice(
             )
 
     @patch('kiwi.repository.dnf.RpmDataBase')
-    def test_setup_package_database_configuration(self, mock_RpmDataBase):
+    @patch('kiwi.command.Command.run')
+    @patch('kiwi.repository.zypper.Path.create')
```
Comment (Conan-Kudo) [inline test/unit/repository/dnf_test.py:167]: Why are we pulling in the Zypper module here?

**Review by Conan-Kudo — CHANGES_REQUESTED** [review]

File: test/unit/repository/dnf_test.py (hunk 1 above)
Comment (davidcassany) [reply test/unit/repository/dnf_test.py:167]: ouch my bad (copy & paste leftover), wondering how it can be that the test passes :thinking:

Comment (Conan-Kudo) [reply test/unit/repository/dnf_test.py:167]: Luck? There are two instances of this I see in this diff, so you probably want to fix that and make sure the tests pass still...

Comment (davidcassany) [reply test/unit/repository/dnf_test.py]: fixed

**Review by Conan-Kudo — APPROVED** [review]

**Pull request merged**